.github/workflows/go-validation.yml

```
@@ -0,0 +1,70 @@
+name: Go Validation
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "go/*"
+      - "go.work"
+      - "go.work.sum"
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "go/*"
+      - "go.work"
+      - "go.work.sum"
+
```

it would be good to add `permissions` at the top level since this is net-new:
```suggestion
permissions:
  contents: read
```
I think just `contents: read` should suffice in this case since this workflow just needs to be able to check out the repository.